imap/index.c

```diff
@@ -5820,32 +5820,42 @@ static int getsearchtext_cb(int isbody, charset_t charset, int encoding,
             goto done;
         }
 
-        /* Extract text from attachment */
-        struct attachextract_record axrecord = {
-            .type = type, .subtype = subtype,
-            .guid = message_guid_clone(content_guid),
-        };
-
-        r = attachextract_extract(&axrecord, data, encoding, &text);
-
-        if (!r && buf_len(&text)) {
-            /* Append extracted text */
-            str->receiver->begin_part(str->receiver, SEARCH_PART_ATTACHMENTBODY);
-            str->receiver->append_text(str->receiver, &text);
-            str->receiver->end_part(str->receiver, SEARCH_PART_ATTACHMENTBODY);
-        }
-        else if (r) {
-            syslog(LOG_ERR, "IOERROR index: can't extract attachment %s (%s/%s): %s",
+        if (str->flags & INDEX_GETSEARCHTEXT_NOCALL_ATTACHEXTRACT) {
+            syslog(LOG_DEBUG, "attachextract disabled for "
+                    "attachment %s (%s/%s)",
                     message_guid_encode(content_guid),
-                    type, subtype, error_message(r));
+                    type, subtype);
 
-            if (str->flags & INDEX_GETSEARCHTEXT_PARTIALS) {
-                /* mark message as partially indexed and continue */
-                str->indexlevel |= SEARCH_INDEXLEVEL_PARTIAL;
-                r = 0;
+            str->indexlevel |= SEARCH_INDEXLEVEL_PARTIAL;
```

Comment:
Is it okay that we're setting it as partial here without checking that partials are allowed?  Ahh, I see a comment later explaining that it is okay.